File: Core/Tests/NSURLQueryDictionaryTests.m
//  NSURLQueryDictionaryTests.m
//
//  Copyright 2014 Awful Contributors. CC BY-NC-SA 3.0 US https://github.com/Awful/Awful.app

@import XCTest;
#import "NSURLQueryDictionary.h"

@interface NSURLQueryDictionaryTests : XCTestCase

@end

@implementation NSURLQueryDictionaryTests

- (void)testSome
{
    NSURL *URL = [NSURL URLWithString:@"?g=hello&what=updog"];
    XCTAssertEqualObjects(QueryDictionaryWithURL(URL), (@{ @"g": @"hello", @"what": @"updog" }));
}

- (void)testOne
{
    NSURL *URL = [NSURL URLWithString:@"?sam=iam"];
    XCTAssertEqualObjects(URL.queryDictionary, (@{ @"sam": @"iam" }));
}

- (void)testOneSkippingOne
{
    NSURL *URL = [NSURL URLWithString:@"?&howdy=maam"];
    XCTAssertEqualObjects(URL.queryDictionary, (@{ @"howdy": @"maam" }));
}

- (void)testEmptyValue
{
    NSURL *URL = [NSURL URLWithString:@"?whodat="];
    XCTAssertEqualObjects(URL.queryDictionary, (@{ @"whodat": @"" }));
}

- (void)testEmptyKey
{
    NSURL *URL = [NSURL URLWithString:@"?=ahoy"];
    XCTAssertEqualObjects(URL.queryDictionary, (@{ @"": @"ahoy" }));
}

- (void)testNoEquals
{
    NSURL *URL = [NSURL URLWithString:@"?hooray"];
    XCTAssertEqualObjects(URL.queryDictionary, (@{ @"hooray": @"" }));
}

- (void)testManyNoEquals
{
    NSURL *URL = [NSURL URLWithString:@"?reach&for&the&sky"];
    XCTAssertEqualObjects(URL.queryDictionary, (@{ @"reach": @"", @"for": @"", @"the": @"", @"sky": @"" }));
}

@end

File: Core/Tests/NSURLQueryDictionaryTests.swift
//  NSURLQueryDictionaryTests.swift
//
//  Copyright 2016 Awful Contributors. CC BY-NC-SA 3.0 US https://github.com/Awful/Awful.app

import AwfulCore
import XCTest

final class NSURLQueryDictionaryTests: XCTestCase {
    func testSome() {
        let url = URL(string: "?g=hello&what=updog")!
        XCTAssertEqual(url.awful_queryDictionary, ["g": "hello", "what": "updog"])
    }
    
    func testOne() {
        let url = URL(string: "?sam=iam")!
        XCTAssertEqual(url.awful_queryDictionary, ["sam": "iam"])
    }
    
    func testOneSkippingOne() {
        let url = URL(string: "?&howdy=maam")!
        XCTAssertEqual(url.awful_queryDictionary, ["": "", "howdy": "maam"])
    }
    
    func testEmptyValue() {
        let url = URL(string: "?whodat=")!
        XCTAssertEqual(url.awful_queryDictionary, ["whodat": ""])
    }
    
    func testEmptyKey() {
        let url = URL(string: "?=ahoy")!
        XCTAssertEqual(url.awful_queryDictionary, ["": "ahoy"])
    }
    
    func testNoEquals() {
        let url = URL(string: "?hooray")!
        XCTAssertEqual(url.awful_queryDictionary, ["hooray": ""])
    }
    
    func testManyNoEquals() {
        let url = URL(string: "?reach&for&the&sky")!
        XCTAssertEqual(url.awful_queryDictionary, ["reach": "", "for": "", "the": "", "sky": ""])
    }
}
